app/models/piece.rb

```diff
@@ -1,14 +1,22 @@
+# rubocop:disable Metrics/ClassLength
 class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
 
   # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/LineLength, Metrics/MethodLength
   def move_to!(new_row, new_col)
-    @piece = Piece.find_by(row_position: new_row, col_position: new_col)
+    Piece.transaction do
+      try_to_move(new_row, new_col)
+      fail ActiveRecord::Rollback if game.determine_check
```

Comment:
If the game is in check, you don't specify what should happen other than the transaction be rolled back.  I'm not sure if anything needs to be done, but the method essentially does nothing in that case.